File: Templates/Go/classes/SimpleClass.go
type ClassName struct {
    property1 Type
    property2 Type
}

func NewClassName(property1 Type, property2 Type) *ClassName {
    return &ClassName{property1, property2}
}

func (c *ClassName) Method1(arg1 Type) {
    // Method logic here
}

func (c *ClassName) Method2(arg1 Type, arg2 Type) {
    // Method logic here
}

File: Templates/Nim/classes/SimpleClass.nim
type
  ClassName = object
    property1: Type
    property2: Type

proc newClassName(property1: Type, property2: Type): ClassName =
  result.property1 = property1
  result.property2 = property2

method method1(obj: var ClassName, arg1: Type) =
  # Method logic here

method method2(obj: var ClassName, arg1: Type, arg2: Type) =
  # Method logic here
